expect.go

```diff
@@ -112,13 +113,18 @@ type Config struct {
 	// custom implementation.
 	WebsocketDialer WebsocketDialer
 
+	// DEPRECATED. Use AssertionHandler instead. Please note that a Formatter
+	// can only be passed with an AssertionHandler. This is provided for compatibility.
+	//
 	// Reporter is used to report failures.
 	// Should not be nil.
 	//
 	// You can use AssertReporter, RequireReporter (they use testify),
 	// or testing.TB, or provide custom implementation.
 	Reporter Reporter
```

Comment:
i'm shared about that: i prefer to avoid giving different "paths" that lead to the same behaviour. having the ability to *continue* using a Reporter when you have an AssertionHandler that does the same.

also, i'm only referencing the field here, not the interface, jic :)

Reply:
I see, makes sense too. Let's deprecate the field then.